core/src/SYCL/Kokkos_SYCL_ParallelFor_Range.hpp

```diff
@@ -94,29 +106,31 @@ class Kokkos::Impl::ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>,
       (void)memcpy_event;
 #endif
 
+      const auto actual_range = policy.end() - policy.begin();
       if (policy.chunk_size() <= 1) {
 #ifdef SYCL_EXT_ONEAPI_AUTO_LOCAL_RANGE
-        const auto actual_range = policy.end() - policy.begin();
         FunctorWrapperRangePolicyParallelForCustom<Functor, Policy> f{
             policy.begin(), functor, actual_range};
         // Round the actual range up to the closest power of two not exceeding
         // the maximum workgroup size
-        const auto max_wgroup_size =
+        const std::size_t max_wgroup_size =
             q.get_device().get_info<sycl::info::device::max_work_group_size>();
-        const auto wgroup_size_multiple = Kokkos::bit_floor(
+        const std::size_t wgroup_size_multiple = Kokkos::bit_floor(
             std::min<std::size_t>(max_wgroup_size, actual_range));
 
-        const auto launch_range = (actual_range + wgroup_size_multiple - 1) /
-                                  wgroup_size_multiple * wgroup_size_multiple;
+        const std::size_t launch_range =
+            (actual_range + wgroup_size_multiple - 1) / wgroup_size_multiple *
+            static_cast<size_t>(wgroup_size_multiple);
```

Comment:
I don't quite understand your choice to cast here versus casting the numerator maybe

Reply:
We shouldn't actually need the cast since it doesn't do anything anymore.
```suggestion
            wgroup_size_multiple;
```